src/main/kotlin/io/emeraldpay/dshackle/upstream/LabelsDetector.kt

```diff
@@ -55,22 +55,8 @@ abstract class BasicEthLabelsDetector(
     }
 
     private fun clientType(client: String): String? {
-        return if (client.contains("erigon", true)) {
-            "erigon"
-        } else if (client.contains("geth", true)) {
-            "geth"
-        } else if (client.contains("bor", true)) {
-            "bor"
-        } else if (client.contains("nethermind", true)) {
-            "nethermind"
-        } else if (client.contains("prysm", true)) {
-            "prysm"
-        } else if (client.contains("lighthouse", true)) {
-            "lighthouse"
-        } else {
-            log.debug("Unknown client type: {}", client)
-            null
-        }
+        val firstSlash = client.indexOf("/")
```

Comment:
What about to add a check of a specific case if `firstSlash` is -1? In that case `substring` can throw an exception